client/templates/globals/profileEditDialog.js

```diff
@@ -31,25 +52,33 @@ Template.profileEditDialog.events({
 
         if (Meteor.user().isDemoUser) { 
             return; 
-        } 
-        
-        let uLongName = tmpl.find('#id_longName').value;
-        let uEmailAddress = tmpl.find('#id_emailAddress').value;
+        }
 
-        tmpl.$('#btnEditProfileSave').prop('disabled',true);
+        let uEmailAddress = tmpl.find('#id_emailAddress').value;
 
-        Meteor.call('users.editProfile', Meteor.userId(), uEmailAddress,uLongName, function (error) {
-            if (error) {
-                tmpl.$('#btnEditProfileSave').prop('disabled',false);
-                console.log(error);
-                showError(evt, error);
-            } else {
-                (new FlashMessage('OK', 'Profile edited.', 'alert-success', 2000)).show();
+        if (Meteor.settings.public.sendVerificationEmail) {
+            let changeUserMail = () => {
+                updateUserProfil(tmpl);
```

Comment:
Possibly. Depends how meteor handles concurrent method calls. If two calls from the same client can run concurrently, this is a race condition. If they are queued, there is none.

If two Meteor methods can run concurrently, the `Meteor.call()` inside `updateUserProfil` might not be finished and still depend on a logged in user before `logout()` is called below.